notebook/notebookapp.py

```diff
@@ -2289,14 +2319,16 @@ def list_running_servers(runtime_dir=None):
     if not os.path.isdir(runtime_dir):
         return
 
+    api_version_json_bytes = json.dumps({"version":__version__}).encode()
```

Comment:
This implies this will only work for like-versioned notebook servers.  For example, if I'm hosting multiple environments that span different NB server versions, any versions running that do not match the version performing the list operation will have their `nbserver.json` files removed due to the mismatch and essentially be "zombied".

Reply:
commit 978d883c8cc3a9e7410eafd0b1cf0c5f2f558944
Author: git <abaelhe@icloud.com>
Date:   Thu Sep 3 21:19:26 2020 +0800

    Multi-instance and multi-version of Jupyter Notebook

commit 07570c67bb6c84194298907aea9253199da2ae3f
Author: Abael He <abaelhe@icloud.com>